### PR TITLE
Improved doll sync

### PR DESCRIPTION
**Pull request by AndreaCatania**
Port of: https://github.com/GodotNetworking/network_synchronizer/pull/30

- Removed the Interpolator: now the interpolation is handled manually.
- Simplified the Interpolation: now the doll doesn't store all the received epoch, it interpolates directly to the most recent one.
- Improved the lag compensation: the algorithm that detects the network oscillations is faster and more accurate.
- Dropped the concept of batch: it will be re-introduced later by batching all the info to send to a specific client.
- Simplified the network settings.
- Improved logging.

### CODE REVIEW

**Comment by AndreaCatania** [comment]
@kubecz3k @aaronfranke Be aware the the doll sync method is now different. Once I merge it you will need to update your repos. Sorry for the non backportable changes.

File: input_network_encoder.cpp [hunk 1]
```diff
@@ -110,19 +110,19 @@ void InputNetworkEncoder::encode(const LocalVector<Variant> &p_input, DataBuffer
 						r_buffer.add_unit_real(pending_input.operator real_t(), info.compression_level);
 						break;
 					case DataBuffer::DATA_TYPE_VECTOR2:
-						r_buffer.add_vector2(pending_input.operator Vector2(), info.compression_level);
+						_ALLOW_DISCARD_ r_buffer.add_vector2(pending_input.operator Vector2(), info.compression_level);
```
Comment (kubecz3k) [inline input_network_encoder.cpp:113]: how _ALLOW_DISCARD_ is working?

**Review by kubecz3k — APPROVED** [review]
Looked at the code, looks good to me, I like how the api is simpler now and at the same time it feels like it gives more power to the user (well it's more explicit how to do custom interpolation now in any case). :+1:

File: input_network_encoder.cpp (hunk 1 above)
Comment (AndreaCatania) [reply input_network_encoder.cpp:113]: `Vector2` is marked as no discard, so you MUST always use the return value. `_ALLOW_DISCARD_` make sure you can just ignore the return type.

**Comment by AndreaCatania** [comment]
Thanks!

**Pull request merged**